platform/linux-generic/pktio/loop.c

```diff
@@ -438,26 +443,68 @@ static inline void loopback_fix_checksums(odp_packet_t pkt,
 		_odp_packet_sctp_chksum_insert(pkt);
 }
 
+static inline odp_queue_t get_dest_queue(const pkt_loop_t *pkt_loop, odp_packet_t pkt, int index)
+{
+	const odp_pktin_hash_proto_t *hash = &pkt_loop->hash;
+	_odp_udphdr_t udp;
+	_odp_tcphdr_t tcp;
+	_odp_ipv4hdr_t ipv4;
+	_odp_ipv6hdr_t ipv6;
+	uint32_t off, dest_hash = 0;
+
+	if (hash->all_bits == 0)
+		return pkt_loop->loopqs[index % pkt_loop->num_qs].queue;
+
+	off = odp_packet_l4_offset(pkt);
+
+	if (off != ODP_PACKET_OFFSET_INVALID) {
+		if ((hash->proto.ipv4_udp || hash->proto.ipv6_udp) && odp_packet_has_udp(pkt)) {
+			if (odp_packet_copy_to_mem(pkt, off, _ODP_UDPHDR_LEN, &udp) == 0)
+				dest_hash ^= odp_be_to_cpu_16(udp.dst_port);
```

Comment:
It would be more user-friendly to use both destination and source addresses for hashing, although this likely requires something else than modulo to work properly.

Reply:
Sure, though as the API doesn't go into detail how granular the per-layer-hashing should be I guess this current solution is fine.

Reply:
One flexible solution could be to use for example `odp_hash_crc32()` over all fields selected for hashing. Assuming the the performance isn't terrible.

Reply:
`odp_hash_crc32()` utilized in v2.